### PR TITLE
21092 Dissolution Job - add logic to initiate dissolution process for businesses (first pass)

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#21092

*Description of changes:*
- Added logic to initiate dissolution process for businesses 
- First pass at unit tests
- This is meant to be an initial PR to unblock others while I iron out some issues with it locally.

*Notes:*

_Edit: fixed!_

I cannot seem to get the requirements installed correctly on my machine, and so I am unable to run the job or tests locally to verify they work. The issue I am getting is:

Screenshot 2024-05-24 at 10 44 48 AM

However, it does install the requirements without issue on Docker, but I get a different error when trying to run it on Docker:

Screenshot 2024-05-24 at 10 45 27 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py
Comment (chenhongjing) [inline jobs/involuntary-dissolutions/involuntary_dissolutions.py:87]: I think it would be better to move this check out of initiate dissolution process(stage 1) because there will also be stage 2 and stage 3 checks to see if they need to run today. They can be combined together. 🤔

Comment (leodube-aot) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py:87]: Good idea! Do you think it also makes sense to move the "only run once a day" check as well? Or will the other stages run multiple times a day?

Comment (chenhongjing) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py:87]: 🤔 I don't think other stages will need this check as per the tech design and my conversation with Argus... 💭

Comment (leodube-aot) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py:87]: Sounds good, I'll only move the cron check then

**Comment by leodube-aot** [comment]
> @leodube-aot I pulled down your PR and was tinkering a bit.  I was able to get `make seutp` working by making the following updates.

Thanks Argus I'll give this a try to see if it works. If not I think I'll try the setup on my Windows machine to see if that fixes it.

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py
Comment (leodube-aot) [reply jobs/involuntary-dissolutions/involuntary_dissolutions.py:126]: That makes sense to me, I'll move the logic out into its own function.

**Comment by leodube-aot** [comment]
When I run the job locally, these are the entries that were created in the `batche_processing` table:

Screenshot 2024-05-30 at 7 32 43 PM

and in the `batches` table:

Screenshot 2024-05-30 at 7 32 49 PM

I am moving the unit tests work to a second PR after this one is merged as I need the `legal_api` changes in this ticket to be merged first.

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py
Comment (argush3) [inline jobs/involuntary-dissolutions/involuntary_dissolutions.py:109]: Can remove these two TODOs now.  No longer a requirement.

File: jobs/involuntary-dissolutions/involuntary_dissolutions.py
Comment (argush3) [inline jobs/involuntary-dissolutions/involuntary_dissolutions.py:76]: I think I use "Initiate dissolution process for new businesses where AR has not been filed for 2 yrs and 2 months." as the title of stage 1 in my tech design.  That is a little inaccurate are other possible selection criteria.  

Should probably just update this to something like "Initiate dissolution process for new businesses that meet dissolution criteria"

**Comment by argush3** [comment]
>Also, the 2 linting errors are due to the legal_api changes in this PR not being merged yet. Would it be worth splitting these off into a separate PR to resolve the lint errors?

yeah, I think it makes as you suggested, Leo.  To resolve the linting errors in another PR after this one is merged.

**Review by argush3 — APPROVED** [review]
Looks good!

Pre-approving assuming that latest changes in `main` will be pulled in and subsequent linting and unit test PRs to come.

**Pull request merged**
